### PR TITLE
Completed Week 4 Assessment

### PR DESCRIPTION
**Pull request by JLODI**

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -33,9 +74,44 @@ const nums1 = [3, 56, 90, -8, 0, 23, 6]
...
 // b) Create the function that makes the test pass.
 
+// psuedo code:
+// define a function that takes in an array
+// itterates through the array
+// sort through numerical order from least to greatest
+// splice out second element to second to last element
+// return array
+
+const minMaxNum = (array) => {
```
Comment (SunkissedQueen) [inline code-challenges.test.js:102]: I love this approach.

File: code-challenges.test.js [hunk 2]
```diff
@@ -20,9 +20,50 @@ const colors1 = ["purple", "blue", "green", "yellow", "pink"]
...
+// define a function that takes in an array
+// removes the first element in the array
+// assign new empty array
+// assign a temporary array thats a copy of the array
+// itterate through temp
+// takes random element from temp array and pushes to new array
+// removes selected element from temp array
+// return new array
+
+const shuffleArray = (array) => {
```
Comment (SunkissedQueen) [inline code-challenges.test.js:50]: I appreciate the logic and the creative use of the built-in methods

File: code-challenges.test.js [hunk 3]
```diff
@@ -45,5 +121,50 @@ const testArray1 = [3, 7, 10, 5, 4, 3, 3]
...
+// create an empty array
+// itterate through the first array
+// write conditional if value is in the new array
+// if not, push to new array
+// itterate through the second array
+// write conditional if value is in the new array
+//if not, push to new array
+// return new array
+
+const arrayCombine = (array1, array2) => {
```
Comment (SunkissedQueen) [inline code-challenges.test.js:152]: Such a unique approach. Now that you know this approach works, find a way not to repeat the lines in your code block. Let's accomplish the same logic with less lines

File: interview-questions.md [hunk 4]
```diff
@@ -8,51 +8,72 @@ Then, research the question to expand on your answer. Even if you feel you have
 1. What are props in React?
 
   Your answer:
+  "Props" is short for properties, and it's the keyword used in React to refer to the properties being passed from one component to another component. In React, the information in props can only be passed from the parent class to child class. They are also "read only" and can't be changed by a function in another component.
 
-  Researched answer:
+  While learning about React in class, I was able to use "props" as a parameter in a constructor (method that initialize's the object's state) which allowed me to pass information about the current state to another component. In the parent component, app.js for example, I would instantiate the component in the render method and type the syntax in the component as an attribute like attribute name={ this.state.key }
```
Comment (SunkissedQueen) [inline interview-questions.md:13]: Love the way you are answering the questions

File: interview-questions.md [hunk 5]
```diff
@@ -8,51 +8,72 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+  Researched answer:
+  "Props" is a keyword in React used to send data from one component to another. Props can't be changed by any other external function and the data is unidirectional, from parent to child component. Props can send any kind of data into a component when that component is being called. Data can be passed around easily to two or more components in this way.
 
 
 2. What is a DOM event?
 
   Your answer:
+  DOM events are occurrences that are caused by user interaction or browser. Examples of DOM events include: user clicking the mouse button, hitting enter, mousing over an element, when a user presses a key, when a web page has loaded, etc.
+
+   When I was practicing developing the treasure hunt and tic tac toe applications using javascript, I would use the "onClick" DOM event to run a javascript function when an HTML element was clicked which converted a question mark into an emoji.
```
Comment (SunkissedQueen) [inline interview-questions.md:24]: Feel so connected to your journey

File: interview-questions.md [hunk 6]
```diff
@@ -8,51 +8,72 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 5. Ruby has an implicit return. What does that mean?
 
   Your answer:
+    Implicit return in Ruby means that the value of the last line of a method is returned automatically without using a syntax like keyword "return"
 
-  Researched answer:
+    This seems to be another example of Ruby's simplicity and developer-friendliness.
 
+  Researched answer:
```
Comment (SunkissedQueen) [inline interview-questions.md:60]: Yes

File: ruby-challenges.rb [hunk 7]
```diff
@@ -21,7 +32,13 @@
 album3 = 'Abbey Road'
 # Expected output: 'bby Rd'
 
+def no_vowels string
```
Comment (SunkissedQueen) [inline ruby-challenges.rb:35]: Great work! FYI, Ruby allows you to use the delete method with passing all of the characters at once. Delete will remove any matches. `.delete "AEIOUaeiou"` When executing this method, that string does not exist as a substring but more like the set of letters A, E, I, O, U, a, e, i, o, u separately.

File: ruby-challenges.rb [hunk 8]
```diff
@@ -31,3 +48,15 @@
 # Expected output: 'LEARN is not a palindrome'
 is_palindrome3 = 'Rotator'
 # Expected output: 'Rotator is a palindrome'
+
+def palChecker word
+    if word.downcase.reverse == word.downcase
```
Comment (SunkissedQueen) [inline ruby-challenges.rb:53]: Perfect, placing them all in the same case makes them logic work smoothly

**Review by SunkissedQueen — APPROVED** [review]
Jennice, your logic and creativity is amazing. I am very impressed with your interview questions. You are doing very well. Appreciate all you are bringing to web development. Wonderful Work!

File: ruby-challenges.rb (hunk 7 above)
Comment (JLODI) [reply ruby-challenges.rb:35]: I'll take note, thanks Charlean!

**Pull request merged**